xrdp/xrdp_encoder.c

```diff
@@ -77,16 +84,24 @@ xrdp_encoder_create(struct xrdp_mm *mm)
 
     client_info = mm->wm->client_info;
 
+    /* RemoteFX 7.1 requires LAN but GFX does not */
     if (client_info->mcs_connection_type != CONNECTION_TYPE_LAN)
     {
-        return 0;
+        if ((mm->egfx_flags & 3) == 0)
```

Comment:
Where is `3` come from?

Reply:
You know I was wondering that myself (Most of this is code's lineage comes from Jay's original egfx branch that was started many years ago).

I think it's this: In this simplified version of GFX, this flag can only ever be 1 (For H264) or 2 (For Progressive RFX). Anding with 3 means "It is neither 1 nor 2":

0b00 & 3 = 0
0b01 & 3 = 1
0b10 & 3 = 2
0b11 & 3 = 3

So basically, if it's not a LAN connection, and it's not EGFX, drop the connection because the only other use for the encoder is "non-progressive RFX" which requires LAN per the spec.

There's probably a clearer way to write this. I think egfx_flags needs to become an enum, too. Thoughts?

Reply:
I agree. Either enum or define as macro would be clearer.

Reply:
Updated!